### PR TITLE
* DataContext.get_checkpoint now returns a Checkpoint

### PR DESCRIPTION
**Pull request by Aylr**
Changes proposed in this pull request:

This PR removes the return_config parameter to make DataContext.get_checkpoint() do what it says by returning a Checkpoint.
CheckpointConfigs are easy to access from the Checkpoint.config property.

### CODE REVIEW

File: tests/data_context/test_data_context.py [hunk 1]
```diff
@@ -1596,26 +1597,31 @@ def test_get_checkpoint_raises_error_empty_checkpoint(
 def test_get_checkpoint(empty_context_with_checkpoint):
     context = empty_context_with_checkpoint
     obs = context.get_checkpoint("my_checkpoint")
-    assert isinstance(obs.to_json_dict(), dict)
-    assert {
-        "validation_operator_name": "action_list_operator",
+    assert isinstance(obs, Checkpoint)
+    config = obs.config
+    assert isinstance(config.to_json_dict(), dict)
+    assert config.to_json_dict() == {
```
Comment (Aylr) [inline tests/data_context/test_data_context.py:1604]: Also note this was asserting a dictionary without comparison.

**Review by alexsherstinsky — APPROVED** [review]
This is a beautiful PR -- and the simplification makes perfect sense.  The only possible problem is that the methods changed could be used by users.  If there are no use-facing methods that were modified, then this should be merged.  Otherwise, if there is any possibility that the users' existing code will stop working (i.e., due to broken backward compatibility), then we should discuss this first.  Thank you. /cc @roblim

**Comment by Aylr** [comment]
Hmm... We are breaking `DataContext.get_checkpoint()` which in develop returns a `dict`. I had made perhaps an incorrect assumption that we intend to do this. It's also possible I'm missing some context here.

This raises two questions:

1. Did we intend to break that API?
2. The prior API is marked **experimental**, so while it isn't the friendliest thing to do, I do think we are within our definition of experimental to break this, unless the thinking on feature maturity has evolved since summer.

**Comment by alexsherstinsky** [comment]
> Hmm... We are breaking `DataContext.get_checkpoint()` which in develop returns a `dict`. I had made perhaps an incorrect assumption that we intend to do this. It's also possible I'm missing some context here.
> 
> This raises two questions:
> 
> 1. Did we intend to break that API?
> 2. The prior API is marked **experimental**, so while it isn't the friendliest thing to do, I do think we are within our definition of experimental to break this, unless the thinking on feature maturity has evolved since summer.

I am OK with this as long as @roblim is OK with it too; and @jcampbell is welcome to provide any also.

**Comment by roblim** [comment]
Great PR. I'm okay with the changes too. I doubt the changes would affect existing user code, though we should definitely mention it as something that's breaking. I think currently, this is primarily used in the CLI.

**Comment by roblim** [comment]
@Aylr - do the CLI operations that use `load_checkpoint()` still work?

**Pull request merged**